### DISCUSSION TITLE
Github permission issues on deploy in Railway

### DISCUSSION THREAD

**Discussion by cshenoy** [coder/code-server#5416] [Q&A] [answered]
Hi there - I followed the Railway deploy guide today and after some tinkering found out I couldn't clone from an organization private repo. I ended up moving my project to a private repo under my personal account (this one) but am still getting issues during the deploy in Railway. I registered with Railway through github and have authorized full access to all repos. I even changed the `GIT_REPO` env to a personal public one to make sure it wasn't something else (that worked). I've pasted relevant portions of the deploy logs below.

```
[deploy-code-server] RCLONE_DATA is not specified. Files will not persist
Cloning into '/home/coder/project'...
fatal: could not read Username for 'https://github.com';: No such device or address
[deploy-code-server] Starting code-server...
```

Using code-server v4.5.1. I feel like I'm missing something simple here but can't pinpoint it. Would appreciate some help. Thank you

**Comment by lg-nineoneninethreenine** [comment]
This actually happens to me sometimes. Perhaps use the GitHub CLI client to clone from org repos - it could be a permission issue. I fix it by using the CLI for Git control, which includes cloning repos.

**Comment by cshenoy** [comment, marked as answer]
I ended up creating a personal access token and adding to the git https url.

e.g. `https://{token}@github.com/{name}/{repo}.git`

**Reply by jsjoeio** [reply]
did this resolve your issue?

**Reply by cshenoy** [reply]
It did. Is it worth adding to the docs or was my case not common?

**Reply by jsjoeio** [reply]
Seems a bit uncommon. Thanks for posting the solution! That should suffice for now.

**Reply by codewitgabi** [reply]
I think it's actually common though and should be added to the doc

**Comment by airstrike** [comment]
I'm on the same boat here and would encourage adding this to the docs

**Reply by code-asher** [reply]
Is that in this repo?  For the life of me I cannot seem to find anything about Railway here aside from a mention and a link to https://github.com/coder/deploy-code-server, so maybe we need to open an issue there.

**Comment by shirecoding** [comment]
this is still a problem i have to make my repository public else it fails at the snapshot is says object not found! this is quite a critical issue
